src/lib/krad/packet.c

```diff
@@ -397,18 +562,30 @@ krad_packet_decode_request(krb5_context ctx, const char *secret,
     krb5_error_code retval;
 
     retval = decode_packet(ctx, secret, buffer, reqpkt);
-    if (cb != NULL && retval == 0) {
-        for (tmp = (*cb)(data, FALSE); tmp != NULL; tmp = (*cb)(data, FALSE)) {
+    if (retval)
+        return retval;
+
+    /* Verify Message-Authenticator if present. */
+    if (has_pkt_msgauth(*reqpkt)) {
+        retval = verify_msgauth(secret, *reqpkt, pkt_auth(*reqpkt));
```

Comment:
The reason why the null `rsp` mistake in `verify_msgauth()` was never noticed is because this function was called at the wrong place in `krad_packet_decode_request()` (it was verifying potential duplicated request rather than the one being actually decoded).